### PR TITLE
imagenet download and prepare

### PR DESCRIPTION
**Pull request by smurfd**
hi, adding a python version of https://gist.github.com/antoinebrl/7d00d5cb6c95ef194c737392ef7e476a in a separate `imagenet_download.py`

### CODE REVIEW

File: datasets/imagenet_download.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  for co,tarf in enumerate(images):
+    # for each tar file found. Create a folder with its name. Extract into that folder. Remove tar file
+    if Path(Path(__file__).parent.parent / "datasets/imagenet/train" / images[co]).is_file():
+      images[co] = tarf[:-4] # remove .tar from extracted tar files
+      if not os.path.exists(Path(__file__).parent.parent / "datasets/imagenet/train" / images[co]): os.makedirs(Path(__file__).parent.parent / "datasets/imagenet/train" / images[co])
+      imagenet_extract(Path(__file__).parent.parent / "datasets/imagenet/train" / tarf, Path(__file__).parent.parent / "datasets/imagenet/train" / images[co], small=True)
+      os.remove(Path(__file__).parent.parent / "datasets/imagenet/train" / tarf)
+
+if __name__ == "__main__":
+  if not os.path.exists(Path(__file__).parent.parent / "datasets/imagenet"): os.makedirs(Path(__file__).parent.parent / "datasets/imagenet")
```
Comment (geohot) [inline datasets/imagenet_download.py:40]: exist_ok=True

File: datasets/imagenet_download.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  if not os.path.exists(Path(__file__).parent.parent / "datasets/imagenet/val"): os.makedirs(Path(__file__).parent.parent / "datasets/imagenet/val")
+  if not os.path.exists(Path(__file__).parent.parent / "datasets/imagenet/train"): os.makedirs(Path(__file__).parent.parent / "datasets/imagenet/train")
+  download_file("https://raw.githubusercontent.com/raghakot/keras-vis/master/resources/imagenet_class_index.json", Path(__file__).parent.parent / "datasets/imagenet/imagenet_class_index.json")
+  download_file("https://raw.githubusercontent.com/tensorflow/models/master/research/slim/datasets/imagenet_2012_validation_synset_labels.txt", Path(__file__).parent.parent / "datasets/imagenet/imagenet_2012_validation_synset_labels.txt")
+  download_file("https://image-net.org/data/ILSVRC/2012/ILSVRC2012_img_val.tar", Path(__file__).parent.parent / "datasets/imagenet/ILSVRC2012_img_val.tar") # 7GB
```
Comment (geohot) [inline datasets/imagenet_download.py:45]: By default maybe only download the val? And flag to select train also

File: datasets/imagenet_download.py (hunk 1 above)
Comment (smurfd) [reply datasets/imagenet_download.py]: yes, looks alot nicer that way!

File: datasets/imagenet_download.py (hunk 2 above)
Comment (smurfd) [reply datasets/imagenet_download.py:45]: makes sense!

**Comment by smurfd** [comment]
added also the env var to `docs/env_vars.md`

**Comment by geohot** [comment]
Sweet!

**Pull request merged**